helix-term/src/ui/picker.rs

```diff
@@ -118,6 +284,31 @@ impl<T> Picker<T> {
         }
     }
 
+    fn calculate_preview(&mut self, editor: &Editor) {
+        if let Some((path, line)) = self
+            .selection()
+            .and_then(|current| (self.preview_fn)(editor, current))
+            .and_then(|(path, range)| canonicalize_path(&path).ok().zip(Some(range)))
+        {
+            // TODO: store only the doc as key and map of range -> view as value ?
+            // will improve perfomance for pickers where most items are from same file (symbol, goto)
```

Comment:
Ah yeah, using this for goto previews would be kind of cool. Bonus points for using documents already available in `editor.documents`.

Reply:
Ah that's a leftover todo, it uses docs as keys now. Reusing `editor.documents` is a good idea, I'll add it.